### ISSUE TITLE
Local mail server integration on docker deployment 

### ISSUE THREAD

**Issue by citizen010** [btcpayserver/btcpayserver-docker#312]
**Describe the solution you'd like**
It would be a great addition if a local mail server is included on docker deployment.
So "localhost"  - or whatever you like - could be added to "Quick fill setting for..." menu and
"Server Settings -> Email server" could be pre-filled with mandatory parameters from the above.

I'm aware that merchant, probably, has to deal with MX DNS records of it's own domain.

**Describe alternatives you've considered**
Gmail, Yahoo, Office 365 are not sovereign enough.
Mailgun and SendGrid are not free.

**Provide examples**
SMTP Server: localhost
Port: 587
Sender's display name : (pre-filled with store name)
Sender's email address: merchant@domain.name
Login: merchant@domain.name
Password: (strong, not editable, generated pwd) + button to generate new pwd
Enable SSL: yes
Test email: registered user email

**Comment by citizen010** [comment]
Maybe this could be useful
https://hub.docker.com/r/juanluisbaptiste/postfix
As I understand, it doesn't need to deal with MX dns records due the fact that is only an SMTP relay, there's no POP3.

**Comment by citizen010** [comment]
I've create a simple docker image to solve this.
You can see it here:

https://hub.docker.com/r/citizen010/postfix

Tested successfully on my personal instance of BTCPayServer. 
Hope this help.

**Comment by NicolasDorier** [comment]
I think this is something very hard in practice. Like you need to setup DKIM if you plan to avoid spam filters. I remember a talented system administrator that 10 years ago told me I should never try to setup my own mail server, except if you have a professional knowing what he is doing.

**Comment by citizen010** [comment]
These are certainly wise words.
Indeed, I'm working on this very problem since 2 weeks without any appreciable results.
Maybe better listen to the talented sysadmin's words.

**Comment by NicolasDorier** [comment]
I never tried myself, but I know that this is freaking hard.

**Comment by lastcron** [comment]
I agree with the sysadmin that gave you that advise , I've walked down this road myself  and believe me , after figthing with all kind of problems ( antivirus , antispam , ddos prevention , antihacking measures , etc ,etc ) ( setting it up is the easy part ) I woud still pay $5 dollars to google or AWS WorkMail to do the job.

**Comment by mpuz** [comment]
> I've create a simple docker image to solve this. You can see it here:
> 
> https://hub.docker.com/r/citizen010/postfix
> 
> Tested successfully on my personal instance of BTCPayServer. Hope this help.

Working perfectly! You saved me a ton of time! Thanks! No need for googlezon!

**Comment by citizen010** [comment]
You're welcome

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
Il domenica 5 dicembre 2021 11:09 AM, Mike ***@***.***> ha scritto:

**Comment by Waldorf3** [comment]
All those excuses for not providing a mail solution.  I set up mailservers all day long, including DKIM, DMARC, spf and whatnot, it really isn't rocket science, and certainly not a simple outgoing relay.   The problems everybody is whining about is for incoming mail servers. The only problem with outgoing is to make sure it can't be abused, and it really isn't that hard.  In fact, if one was provided only listening on an internal docker port it would be a non issue.

Normally if I run a simple postfix container on the same network as btcpay it should be possible to enter the containers name (postfix) as hostname and internal port (587) and it should work out of the box but instead btcpay fails with a "Error: Name or service not known".   Two containers on the same docker network ought to be able to communicate with each other using the container_name's, why isn't this possible here?

**Comment by pavlenex** [comment]
Feel free to do a pull request and contribute @Waldorf3
